src/main/java/com/minecolonies/coremod/tileentities/TileEntityColonyBuilding.java

```diff
@@ -139,6 +145,55 @@ public BlockPos getPosition()
         return pos;
     }
 
+    /**
+     * Check for a certain item and return the position of the chest containing it.
+     *
+     * @param itemStackSelectionPredicate the stack to search for.
+     * @return the position or null.
+     */
+    @Nullable
+    public BlockPos getPositionOfChestWithItemStack(@NotNull final Predicate<ItemStack> itemStackSelectionPredicate)
+    {
+        @Nullable final AbstractBuilding building = getBuilding();
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Rename "building" which hides the field declared at line 49. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AHiddenFieldCheck)

Reply:
eithor rename this or the field